src/domain/create-entry.ts

```diff
@@ -58,50 +58,55 @@ export class CreateEntryService {
       sourceTemplate.url,
       sourceTemplate.stripPrefix
     );
-    const integrityHash = computeIntegrityHash(releaseArchive.diskPath);
-    sourceTemplate.setIntegrityHash(integrityHash);
 
-    const moduleFile = await releaseArchive.extractModuleFile(moduleRoot);
+    try {
+      const integrityHash = computeIntegrityHash(releaseArchive.diskPath);
+      sourceTemplate.setIntegrityHash(integrityHash);
 
-    const bcrEntryPath = path.resolve(
-      bcrRepo.diskPath,
-      "modules",
-      moduleFile.moduleName
-    );
-    const bcrVersionEntryPath = path.join(bcrEntryPath, version);
+      const moduleFile = await releaseArchive.extractModuleFile(moduleRoot);
 
-    if (!fs.existsSync(bcrEntryPath)) {
-      fs.mkdirSync(bcrEntryPath);
-    }
+      const bcrEntryPath = path.resolve(
+        bcrRepo.diskPath,
+        "modules",
+        moduleFile.moduleName
+      );
+      const bcrVersionEntryPath = path.join(bcrEntryPath, version);
 
-    const metadataTemplate = rulesetRepo.metadataTemplate(moduleRoot);
+      if (!fs.existsSync(bcrEntryPath)) {
+        fs.mkdirSync(bcrEntryPath);
+      }
 
-    updateMetadataFile(metadataTemplate, bcrEntryPath, version);
+      const metadataTemplate = rulesetRepo.metadataTemplate(moduleRoot);
 
-    fs.mkdirSync(bcrVersionEntryPath);
+      updateMetadataFile(metadataTemplate, bcrEntryPath, version);
 
-    this.addPatches(
-      rulesetRepo,
-      sourceTemplate,
-      moduleFile,
-      bcrVersionEntryPath,
-      moduleRoot
-    );
+      fs.mkdirSync(bcrVersionEntryPath);
 
-    this.patchModuleVersionIfMismatch(
-      moduleFile,
-      version,
-      sourceTemplate,
-      bcrVersionEntryPath
-    );
+      this.addPatches(
+        rulesetRepo,
+        sourceTemplate,
+        moduleFile,
+        bcrVersionEntryPath,
+        moduleRoot
+      );
 
-    sourceTemplate.save(path.join(bcrVersionEntryPath, "source.json"));
-    moduleFile.save(path.join(bcrVersionEntryPath, "MODULE.bazel"));
+      this.patchModuleVersionIfMismatch(
+        moduleFile,
+        version,
+        sourceTemplate,
+        bcrVersionEntryPath
+      );
 
-    fs.copyFileSync(
-      rulesetRepo.presubmitPath(moduleRoot),
-      path.join(bcrVersionEntryPath, "presubmit.yml")
-    );
+      sourceTemplate.save(path.join(bcrVersionEntryPath, "source.json"));
+      moduleFile.save(path.join(bcrVersionEntryPath, "MODULE.bazel"));
+
+      fs.copyFileSync(
+        rulesetRepo.presubmitPath(moduleRoot),
+        path.join(bcrVersionEntryPath, "presubmit.yml")
+      );
+    } finally {
+      releaseArchive.cleanup();
```

Comment:
This is only part that changed in this file. The diff is messy because everything went into a try...finally block.